code/modules/mob/living/silicon/ai/ai.dm

```diff
@@ -210,6 +218,7 @@
 	QDEL_NULL(doomsday_device)
 	QDEL_NULL(aiMulti)
 	QDEL_NULL(alert_control)
+	UnregisterSignal(src, COMSIG_MOVABLE_HEAR, PROC_REF(handle_hearing))
```

Comment:
Signals are automatically unregistered on Destroy() but I guess it doesn't hurt to explicitly unregister it.